### PR TITLE
add HttpRequestMock

### PR DESCRIPTION
**Pull request by xieofxie**
It is not a trivial change, so design should be clarified first.
Changes:
* In TestScript, `List<HttpRequestMock> HttpRequestMocks` is added for the mocks
    - HttpRequestMock is an interface, so they could also be loaded from files
    - The list will be matched from begin to end and the first matched mock will be used
    - HttpRequestSequenceMock is added as the first type of mock. It will return a list of responses in sequence
* MockHttpRequestMiddleware will set mock HttpClient into turnstate
* Add MockHttpRequest which inherits HttpRequest and will use HttpClient from TurnState
* In HttpRequest, when option is HttpClient, it will be used

Question:
* Should I overwrite HttpRequest in TestingComponentRegistration or ResourceExplorer().RegisterType?

Follow up if the design is appliable:
* Add more properties to HttpRequestSequenceMock, HttpResponseMock

I also tested this for https://github.com/microsoft/botbuilder-dotnet/pull/3813
That pr will not work because when response is saved to memory scope like dialog, the byte array will be serialized to string.

Thanks~

### CODE REVIEW

**Comment by xieofxie** [comment]
Add schemas and fix some.

**Comment by xieofxie** [comment]
Force push to fix tests/tests.schema merge..
It is a good to put HttpClient in TurnState as it will be easy for custom actions to setup their mock.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/TestScriptTests/TestScriptTests_HttpRequestMock.test.dialog [hunk 1]
```diff
@@ -0,0 +1,115 @@
+﻿{
+    "$schema": "../../../tests.schema",
+    "$kind": "Microsoft.Test.Script",
+    "description": "Test HttpRequestMock",
+    "httpRequestMocks": [
+        "get.mock",
+        "all.mock"
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/TestScriptTests/TestScriptTests_HttpRequestMock.test.dialog:7]: very cool!

**Comment by cleemullins** [comment]
@xieofxie, can you resolve the merge conflicts in the tests.schema file? Then we can merge.

**Comment by xieofxie** [comment]
@cleemullins resolved. It also updates schema for Microsoft.QnAMakerDialog.

**Pull request merged**